From d234c6d065b9723c1dc620a1f122e0b56d45efea Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Fri, 10 May 2024 15:21:50 +0000
Subject: [PATCH] edk2-firmware-tegra: Allow UEFI capsule downgrades

    In the case of a rollback to 35.4.1 from 35.5.0 we need
    to rollback to the old bootloader, and without this change
    the downgrade is not possible because the firmware checks
    the capsule versions.

Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 FmpDevicePkg/FmpDxe/FmpDxe.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/FmpDevicePkg/FmpDxe/FmpDxe.c b/FmpDevicePkg/FmpDxe/FmpDxe.c
index 1e7ec4a09e..bf42c8e706 100644
--- a/FmpDevicePkg/FmpDxe/FmpDxe.c
+++ b/FmpDevicePkg/FmpDxe/FmpDxe.c
@@ -982,13 +982,15 @@ CheckTheImageInternal (
   if (Version < Private->Descriptor.LowestSupportedImageVersion) {
     DEBUG (
       (DEBUG_ERROR,
-       "FmpDxe(%s): CheckTheImage() - Version Lower than lowest supported version. 0x%08X < 0x%08X\n",
+       "FmpDxe(%s): CheckTheImage() - Version Lower than lowest supported version. 0x%08X < 0x%08X - continuing anyway\n",
        mImageIdName, Version, Private->Descriptor.LowestSupportedImageVersion)
       );
+    /*
     *ImageUpdatable    = IMAGE_UPDATABLE_INVALID_OLD;
     Status             = EFI_SUCCESS;
     *LastAttemptStatus = LAST_ATTEMPT_STATUS_DRIVER_ERROR_VERSION_TOO_LOW;
     goto cleanup;
+    */
   }
 
   //
-- 
2.17.1

